transports/quic/src/transport.rs

```diff
@@ -324,8 +331,8 @@ impl Listener {
             match ready!(if_watcher.poll_if_event(cx)) {
                 Ok(IfEvent::Up(inet)) => {
                     let ip = inet.addr();
-                    if self.endpoint.socket_addr.is_ipv4() == ip.is_ipv4() {
-                        let socket_addr = SocketAddr::new(ip, self.endpoint.socket_addr.port());
+                    if self.endpoint.socket_addr().is_ipv4() == ip.is_ipv4() {
+                        let socket_addr = SocketAddr::new(ip, self.endpoint.socket_addr().port());
```

Comment:
This could also delegate to `Endpoint` for the benefit of slightly simpler code here. The conversion to a mutliaddress could also live there, thus reducing duplication between the `Up` and `Down` branches.

Reply:
What do you think of e393fe5? I would consider this transport / listener-related logic, thus I kept it as a separate function within the transport mod.

Reply:
LGTM!